data/mods/gen1stadium/scripts.ts

```diff
@@ -529,15 +528,40 @@ export const Scripts: ModdedBattleScriptsData = {
 			basePower = this.battle.clampIntRange(basePower, 1);
 
 			// We now check attacker's and defender's stats.
-			let level = pokemon.level;
-			let attacker = pokemon;
-			const defender = target;
-			if (move.useTargetOffensive) attacker = target;
-			let atkType: StatIDExceptHP = (move.category === 'Physical') ? 'atk' : 'spa';
-			const defType: StatIDExceptHP = (move.defensiveCategory === 'Physical') ? 'def' : 'spd';
-			if (move.useSourceDefensiveAsOffensive) atkType = defType;
-			let attack = attacker.getStat(atkType);
-			let defense = defender.getStat(defType);
+			let level = source.level;
+			const attacker =
+				(move.useBaseOffensiveStatAndBoosts && move.useBaseOffensiveStatAndBoosts.includes('target')) ? target : source;
+			const defender =
+				(move.useBaseDefensiveStatAndBoosts && move.useBaseDefensiveStatAndBoosts.includes('source')) ? source : target;
```

Comment:
```suggestion
			const attacker = move.useBaseOffensiveStatAndBoosts?.includes('target') ? target : source;
			const defender = move.useBaseDefensiveStatAndBoosts?.includes('source') ? source : target;
```

Reply:
Done. Didn't know that existed, very cool!